app/controllers/staff/activity_forms_controller.rb

```diff
@@ -31,12 +30,28 @@ def update
     @activity = Activity.find(params[:activity_id])
     authorize @activity
 
-    @activity.assign_attributes(activity_params)
-    update_wizard_status
+    begin
+      @activity.assign_attributes(activity_params)
+    rescue ActiveRecord::MultiparameterAssignmentErrors
+      assign_error_messages_to_record
+      return render_wizard
+    end
 
+    update_wizard_status
     render_wizard @activity
   end
 
+  def assign_error_messages_to_record
+    filtered_params = activity_params.select { |param| activity_params[param] == "0" }
```

Comment:
Is this going to also catch e.g. a code of `0`? Would feel safer if we explicitly checked the date params, perhaps before doing the assignment. However, doesn't this make more sense to be on the `Activity`? Validate date fields to not contain `0` date components?

Reply:
Oh. I read your description again. I see why it's not on the model. Does the error occur on assignment or on save?

Reply:
On assignment, full trace here:

```
1) Users can create a fund level activity when the user belongs to BEIS validations validation errors work as expected
     Failure/Error: @activity.assign_attributes(activity_params)
     
     ActiveRecord::MultiparameterAssignmentErrors:
       2 error(s) on assignment of multiparameter attributes [error on assignment [1, 0, 2010] to planned_start_date (mon out of range),error on assignment [0, 12, 2010] to planned_end_date (mday out of range)]
     # ./app/controllers/staff/activity_forms_controller.rb:34:in `update'
     # ./spec/features/staff/users_can_create_a_fund_level_activity_spec.rb:134:in `block (4 levels) in <main>'
     # ./spec/rails_helper.rb:73:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:72:in `block (2 levels) in <top (required)>'
```

Reply:
In *theory* entering 0 into other params should be OK, as they won't cause `ActiveRecord::MultiparameterAssignmentErrors` and we won't get here. But I agree, it's fragile as hell and I think this commit is a bad idea. But I wanted to see if i could try to get the error messages on the right fields, and not just one generic error message.

Reply:
Perhaps selecting only fields that have a `_date` suffix and reraising otherwise would make it more robust. I think it would also help clarify what's happening here.